File: src/main/java/com/tang/designpatterns/chap2/java/PrintBanner.java
package com.tang.designpatterns.chap2.java;


/**
 * @Title: PrintBanner
 * @Description:
 * @author: tangyao
 * @date: 2022/9/28 14:48
 * @Version: 1.0
 */

public class PrintBanner extends Banner implements Print {

    public PrintBanner(String s) {
        super(s);
    }

    @Override
    public void printWeak() {
        showWithParen();
    }

    @Override
    public void printStrong() {
        showWithAster();
    }
}

File: src/main/java/com/tang/designpatterns/chap2/scala/PrintBanner.scala
package com.tang.designpatterns.chap2.scala

/**
 * 适配器模式 一个接口，一个类
 *
 * @param string
 */
class PrintBanner(override val string: String) extends Banner(string), Print {

  override def printStrong(): Unit = showWithAstar()

  override def printWeak(): Unit = showWithParen()
}

@main def main = {
  val printBanner = new PrintBanner("hello")

  printBanner.printStrong()
  printBanner.printWeak()

}